### PR TITLE
Introduce new CompensationSubscription record

### LINKED ISSUE #15063: Introduce new compensation subscription record

### LINKED ISSUE #15140: Export compensation subscription events
## Description

We introduced a new compensation record for BPMN compensation events. We should export these events to Elasticsearch and OpenSearch. 

We should follow the development guide on how to add the record for the exporters [here](https://github.com/camunda/zeebe/blob/main/docs/developer_handbook.md#support-a-recordvalue-in-the-elasticsearch-exporter).

requires #15063

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

With this PR a new record CompensationSubscription is added into the protocol

## Related issues

closes #15063 
closes #15140

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]
@nicpuppa nice work. :tada: 

I have one minor comment about the record property names. They don't align well with existing property names, for example, in `ProcessInstanceRecordValue`, `ProcessEventRecordValue`, and `SignalSubscriptionRecordValue`.

But finding names is difficult. I made some suggestions. Please have a look if you agree. :cookie:

**Comment by nicpuppa** [comment]
Thanks @saig0 for the review, naming is always hard 🤣 Please have another look 👀

File: protocol/src/main/java/io/camunda/zeebe/protocol/record/value/CompensationSubscriptionRecordValue.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  long getProcessInstanceKey();
+
+  /**
+   * @return the key of the process definition
+   */
+  long getProcessDefinitionKey();
+
+  /**
+   * @return the id of the element activity with compensation handler
+   */
```
Comment (saig0) [inline protocol/src/main/java/io/camunda/zeebe/protocol/record/value/CompensationSubscriptionRecordValue.java:44]: :wrench: Correct JavaDoc.

```suggestion
  /**
   * @return the element id of the activity with the compensation handler
   */
```

File: protocol/src/main/java/io/camunda/zeebe/protocol/record/value/CompensationSubscriptionRecordValue.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  String getCompensableActivityId();
+
+  /**
+   * @return the flow scope key of the element activity with compensation handler
+   */
```
Comment (saig0) [inline protocol/src/main/java/io/camunda/zeebe/protocol/record/value/CompensationSubscriptionRecordValue.java:49]: :wrench: Correct JavaDoc.

```suggestion
  /**
   * @return the element id of the flow scope that contains the activity with the compensation handler
   */
```

**Review by saig0 — APPROVED** [review]
@nicpuppa LGTM. :tada:

**Pull request merged**